lib/rules/no-extra-parens.js

```diff
@@ -422,7 +438,10 @@ module.exports = function(context) {
             });
         },
         "ReturnStatement": function(node) {
```

Comment:
You can call `context.getSourceCode()` once before all of these functions and then just use the reference in each function rather than repeating this in each function.